spec.bs

```diff
@@ -3362,69 +3432,75 @@ methods for event-level <dfn>forDebuggingOnly reports</dfn> for winning and losi
-          [=bid debug reporting info/seller debug loss report url=], |bidDebugReportInfo|'s
-          [=bid debug reporting info/component seller=], |auctionReportInfo|'s
+          [=bid debug reporting info/bidder debug loss report url=], |bidDebugReportInfo|'s
+          [=bid debug reporting info/interest group owner=], and |auctionReportInfo|'s
           [=auction report info/debug loss report urls=].
-        1. [=Collect a single forDebuggingOnly report=] with |bidDebugReportInfo|'s
-          [=bid debug reporting info/top level seller debug loss report url=], |seller|, and
-          |auctionReportInfo|'s [=auction report info/debug loss report urls=].
+        1. If |bidDebugReportInfo|'s [=bid debug reporting info/component seller=] is null:
+          1. [=Collect a single forDebuggingOnly report=] with |bidDebugReportInfo|'s
+            [=bid debug reporting info/seller debug loss report url=], |seller|, and
+            |auctionReportInfo|'s [=auction report info/debug loss report urls=].
+        1. Otherwise:
+          1. [=Collect a single forDebuggingOnly report=] with |bidDebugReportInfo|'s
+            [=bid debug reporting info/seller debug loss report url=], |bidDebugReportInfo|'s
+            [=bid debug reporting info/component seller=], |auctionReportInfo|'s
+            [=auction report info/debug loss report urls=].
+          1. [=Collect a single forDebuggingOnly report=] with |bidDebugReportInfo|'s
+            [=bid debug reporting info/top level seller debug loss report url=], |seller|, and
+            |auctionReportInfo|'s [=auction report info/debug loss report urls=].
   1. Return |auctionReportInfo|.
 </div>
 
-<div>
+<div algorithm>
   To <dfn>register bids for forDebuggingOnly reports</dfn> given a [=list=] of
-  [=generated bids=] |generatedBids|, [=bid debug reporting info=] |bidDebugReportInfo|,
-  a [=list=] of [=bid debug reporting info=] |bidDebugReportInfoList|:
-  1. Let |mainId| be |bidDebugReportInfoList|'s [=list/size=].
-  1. Let |subId| be 0.
-  1. [=list/Append=] |bidDebugReportInfo| to |bidDebugReportInfoList|.
+  [=generated bids=] |generatedBids|, [=interest group=] |ig|, [=bid debug reporting info=]
+  |bidDebugReportInfo| and a [=reporting context=] |reportingContext|:
+  1. Let |id| be a new [=reporting bid key=] with the following [=struct/items=]:
+    : [=reporting bid key/context=]
+    :: |reportingContext|
+
+    : [=reporting bid key/source=]
+    :: [=reporting bid source/generate-bid=]
+
+    : [=reporting bid key/origin=]
+    :: |ig|'s [=interest group/owner=]
+
+    : [=reporting bid key/interest group name=]
+    :: |ig|'s [=interest group/name=]
+  1. [=map/Set=] |reportingContext|'s [=reporting context/debug reporting info=] [|id|] to
+    |bidDebugReportInfo|.
   1. [=list/For each=] |generatedBid| of |generatedBids|:
-    1. If |generatedBid|'s [=generated bid/for k-anon auction=] is true:

```

Comment:
...I need to figure out how to not report the non-k-anon run properly, I think it's wrong right now.

Reply:
Done'ish. Also fixed additional bids not participating in non-k-anon auction.